install.sh

```
@@ -1129,16 +1129,175 @@ step_openclaw_setup() {
+  fi
+  # A no-op when the shim is absent; a stale shim must never outlive its agent.
+  rm -f "$shim"
+
   echo "  Installing Hermes agent (NousResearch)..."
   # Official installer clones NousResearch/hermes-agent + builds a venv. Runs as
   # the clawbox user so it lands in ~/.hermes and ~/.local/bin.
+  # The URL is passed as an argument rather than spliced into the -c string, so
+  # it stays a single source of truth without shell-quoting exposure.
   runuser -u "$CLAWBOX_USER" -- bash -c \
-    'curl -fsSL https://hermes-agent.nousresearch.com/install.sh | bash' \
+    'curl -fsSL "$1" | bash' _ "$installer_url" \
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)